streets_reader/src/extract.rs

```diff
@@ -53,10 +52,21 @@ impl OsmExtract {
             self.cycleway_stop_lines.push((node.pt.to_hashable(), dir));
         }
 
-        // TODO Maybe restricting to traffic_signals is too much. But we definitely don't want to
-        // use crossing=unmarked to infer stop lines
-        if node.tags.is("highway", "crossing") && node.tags.is("crossing", "traffic_signals") {
-            self.signalized_crossings.push(node.pt.to_hashable());
+        if node.tags.is("highway", "crossing") || node.tags.is("railway", "crossing") {
+            let kind = match node.tags.get("crossing").map(|x| x.as_str()) {
+                Some("traffic_signals") => CrossingKind::Signalized,
+                Some("uncontrolled" | "marked") => CrossingKind::Marked,
```

Comment:
Based on my understanding of https://wiki.openstreetmap.org/wiki/Tag:highway=crossing?uselang=en

Reply:
Kind of. The [history around crossing classification](https://wiki.openstreetmap.org/wiki/Crossings#Street_crossings) is quite muddled and quickly devolves into a blame game. Some mappers strongly believe `uncontrolled` should mean marked/signposted/unsignalized, while others point out that `uncontrolled` in some regions such as the UK and U.S. was historically largely used on unmarked crossings due to plain language in both dialects. I think it’s fair to say that the former faction is winning, but only because the latter has moved on to `crossing:markings=*` and `crossing:signals=*`. At this point, a new renderer or router would be better off supporting `crossing:markings=*` and `crossing:signals=*`, which are less ambiguous in practice.